Subject: [PATCH] fix NPE in TextUtils

return false if String is null or empty. Fix NPE too.

see https://code.google.com/p/android/issues/detail?id=24965

diff --git a/core/java/android/text/TextUtils.java b/core/java/android/text/TextUtils.java
index 48bb5dd..64dd3a1 100644
--- a/core/java/android/text/TextUtils.java
+++ b/core/java/android/text/TextUtils.java
@@ -1474,6 +1474,9 @@ public static boolean isGraphic(char c) {
      * Returns whether the given CharSequence contains only digits.
      */
     public static boolean isDigitsOnly(CharSequence str) {
+        if (isEmpty(str)) {
+            return false;
+        }
         final int len = str.length();
         for (int cp, i = 0; i < len; i += Character.charCount(cp)) {
             cp = Character.codePointAt(str, i);
